include/http/request_handler.js

```diff
@@ -144,6 +145,87 @@ module.exports = function RequestHandlerModule(pb) {
 
     /**
      * @static
+     * @method saveRedirectHost
+     * @param {String} hostname
+     * @param {String} oldHostname
+     * @param {String} uid
+     */
+    RequestHandler.saveRedirectHost = function(hostname, oldHostname, uid) {
+        var self = this;
+        RequestHandler.redirectHosts[hostname] = uid;
+        RequestHandler.sites[oldHostname] = null;
+
+        var dao = new pb.DAO();
+        // If a 301 loop is detected, delete needless redirect
+        if (RequestHandler.redirectHosts[oldHostname]) {
+            self.deleteRedirectHost(oldHostname);
+        }
+
+        var redirect = {host: hostname,
+                        uid: uid};
+        var redirectHost = pb.DocumentCreator.create('redirect_hosts', redirect);
+        dao.save(redirectHost, function(err, result) {
+            if (pb.util.isError(err)) {
+                pb.log.error("ERROR: Failed to save off redirectHost ERR[" + err.stack + "]");
+
+            }
+        });
+    };
+
+    /**
+     * @static
+     * @method deleteRedirectHost
+     * @param {String} hostname
+     */
+    RequestHandler.deleteRedirectHost = function(hostname) {
+        var dao = new pb.DAO();
+        dao.delete({host:hostname}, 'redirect_hosts', function(err, result) {
+            if (pb.util.isError(err)) {
+                pb.log.error("ERROR: Failed to delete deprecated redirectHost ERR[" + err.stack + "]");
+            }
+        });
+    };
+
+    /**
+     * @static
+     * @method checkForRedirect
+     * @param {String} hostname
+     * @param {Function} cb
+     */
+    RequestHandler.checkForRedirect = function(hostname, cb) {
+        var err = null,
+            siteObj = RequestHandler.sites[hostname];
+
+        if (siteObj) {
+            return cb(err, siteObj);
+        }
+
+        else {
+            var redirectSite = RequestHandler.redirectHosts[hostname];
+            if (redirectSite) {
+                new pb.SiteService().getAllSites(function (err, results) {
```

Comment:
You definitely do not want to do this.  A DB call on every request is extremely heavy.  If you have the cached site keep track of its previous hostnames you eliminate the DB call and the possibility of redirecting N number of times until you get to the latest name.  

When a site is "loaded" it should take its array of previous host names and register then in a lookup hash of old hostname to new hostname.  When the request comes in you take the hostname and compare it to the hash of "old" hostnames.  If you get a match then replace the hostname in the request URL and redirect.

This limits the data changes to just the site service's merge function:
```
var dto = ...;
var entity = ...;
entity.prevHostNames = util.isArray(entity.prevHostNames) ? entity.prevHostnames : [];

//when a new hostname is provided (excluding on creation) we add the current hostname to the prev list before assigning the new hostname
if (dto.hostname !== entity.hostname && !!entity.hostname) {
  if (entity.prevHostnames,indexOf(dto.hostname) == -1) {
    entity.prevHostNames.push(entity.hostname);
  }
}
entity.hostname = dto.hostname;
```